### PR TITLE
Flush Ribbon Lend RBN Reward to Ribbon Earn Depositors

### PR DESCRIPTION
**Pull request by chuddster**

### CODE REVIEW

File: contracts/vaults/EarnVault/RibbonEarnVault.sol [hunk 1]
```diff
@@ -1150,6 +1158,33 @@ contract RibbonEarnVault is
...
+
+        // For every Ribbon Lend pool allocated to, withdraw RBN reward from
+        // ribbon lend factory
+        IRibbonLendFactory(0x312853485a41f76f20A14f927Cd0ea676588936C)
+            .withdrawReward(pools);
+
+        IERC20 RBN = IERC20(0x6123B0049F904d730dB3C36a31167D9d4121fA6B);
+        // Transfer RBN to minter
+        RBN.transfer(
+            0x5B0655F938A72052c46d2e94D206ccB6FF625A3A,
```
Comment (kenchangh) [inline contracts/vaults/EarnVault/RibbonEarnVault.sol:1183]: Let's make these addresses constant?

File: contracts/vaults/EarnVault/RibbonEarnVault.sol [hunk 2]
```diff
@@ -1150,6 +1158,33 @@ contract RibbonEarnVault is
...
+    function _withdrawRibbonLendRBNReward() internal {
+        uint256 ribbonLendBorrowerIndex = Vault.RIBBON_LEND_BORROWER_INDEX;
+        uint256 borrowersLength = borrowers.length;
+
+        address[] memory pools =
+            new address[](borrowersLength - ribbonLendBorrowerIndex);
+
+        for (uint256 i = ribbonLendBorrowerIndex; i < borrowersLength; i++) {
+            pools[i - ribbonLendBorrowerIndex] = borrowers[i];
+        }
```
Comment (kenchangh) [inline contracts/vaults/EarnVault/RibbonEarnVault.sol:1173]: What if we just had a one time migration to remove the initial borrowers out of the borrower basket? Then we don't have to deal with these special cases around `RIBBON_LEND_BORROWER_INDEX`, which makes it quite prone to error

File: contracts/vaults/EarnVault/RibbonEarnVault.sol (hunk 1 above)
Comment (chuddster) [reply contracts/vaults/EarnVault/RibbonEarnVault.sol:1177]: It is only used here...think it is fine.

File: contracts/vaults/EarnVault/RibbonEarnVault.sol (hunk 2 above)
Comment (chuddster) [reply contracts/vaults/EarnVault/RibbonEarnVault.sol]: yes, I agree. updated

**Review by kenchangh — APPROVED** [review]

**Pull request merged**